includes/class-wc-payments-account.php

```diff
@@ -225,20 +225,14 @@ private function should_redirect_to_onboarding() {
 			return false;
 		}
 
-		// Don't redirect if the user is in the WC setup wizard.
-		$current_page = isset( $_GET['page'] ) ? wp_unslash( $_GET['page'] ) : null; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-		if ( 'wc-setup' === $current_page ) {
-			return false;
-		}
-
-		// Don't redirect if the user is on the WC-Admin setup profiler or WC-Admin dashboard with the task list.
-		if ( 'wc-admin' === $current_page && empty( $_GET['path'] )
-			&& ( Onboarding::should_show_profiler() || Onboarding::should_show_tasks() ) ) {
-			return false;
-		}
-
-		// Don't redirect if the user is on Jetpack pages.
-		if ( 'jetpack' === $current_page ) {
+		// Do not redirect if the plugin was installed via the WC setup wizard.
```

Comment:
This will still redirect if the plugin is installed as part of the dashboard Task List, and the user navigates away from the Dashboard. Should we just not redirect, ever, unless the current request is a `plugin_activation` request? I think that would simplify this mess a lot.

Reply:
This is great! I ❤️  seeing all that logic removed. This is much simpler and more robust. :shipit: